src/solvers/solver.jl

```diff
@@ -64,12 +64,13 @@ decorate the [`AbstractManifoldObjective`](@ref)` o` with specific decorators.
 optional arguments provide necessary details on the decorators.
 A specific one is used to activate certain decorators.
 
-* `cache` – (`missing`) specify a cache. Currenlty `:Simple` is supported and `:LRU` if you
+* `cache`     – (`missing`) specify a cache. Currenlty `:Simple` is supported and `:LRU` if you
   load `LRUCache.jl`. For this case a tuple specifying what to cache and how many can be provided,
   i.e. `(:LRU, [:Cost, :Gradient], 10)`, where the number specifies the size of each cache.
   and 10 is the default if one omits the last tuple entry
-* `count` – (`missing`) specify calls to the objective to be called, see [`ManifoldCountObjective`](@ref) for the full list
-
+* `count`     – (`missing`) specify calls to the objective to be called, see [`ManifoldCountObjective`](@ref) for the full list
+* `objective` - (`missing`) specify that an objective is `:Euclidean`, which is equivalent to specifying the default embedding
+                the equivalent statement here is `:Embedded`, which fits better in naming when `M` is an []`
```

Comment:
That's not a particularly clear explanation IMO. What `[]` was supposed to be? Why have both `:Euclidean` and `:Embedded` when there is no difference?

Reply:
Well-spotted – I was maybe already in the phase of “Let's first see whether the idea is good”. I meant of course to refer to the `EmbeddedManifold`, since this wrapper would not only work for Euclidean embeddings but for any

Reply:
`objective_domain` sounds more precise, that is correct and “missing” would mean “manifold/inherit”, we could maybe even go to `:Manifold` as default then.

Reply:
The current amount of documentation was mainly done for me to structure (and not forget) my own ideas/thoughts here ;)

Reply:
OK, `:manifold` sounds fine as well.

Reply:
I went for `:Riemannian` as compared to `:Euclidean` in the symbol here.

Reply:
OK, and what about `objective_domain`?

Reply:
Oh sorry, forget that. Will do that after my next meeting :) 

Though with `Riemannian` and `Euclidean` objective itself would actually also make sense already, but you are right, domain is more precise here.

Reply:
Hm, now that I do use `:Riemannian`, `:Euclidean` as symbols, I feel that (similar to gradient and hessian) just stating it is a Riemanian/Euclidean objective should be fine enough (without domain) and still be precise enough.

A Riemannian objective is one that has a Riemannian cost/grad/Hessian
A Euclidean objective has a Euclidean cost/grad/objective (where to be more precise the cost just needs and embed for p before calling the inner cost).

With that, do you thing `objectvie=:Riemannian` would be fine enough?

Reply:
I still find it a bit weird that there is the actual objective passed an argument and an argument called `objective` that isn't it.

Reply:
So, maybe just call it `domain`? After all, it applies to objective, its gradient and Hessian (when present).

Reply:
That sounds reasonable, but for me domain would be a set and miss the information whether its Euclidean or not? `domain=:Euclidean` and `domain=:Riemannian` sound/read a bit strange?

Reply:
Hm... what about `embedded::Bool=false`?

Reply:
I like that one better, just that my idea of making it easier for Manopt(Matlab) people of thinking of the Euclidean objective vanishes in this bool's name. I am also not sure whether we might have other things we want to “transport” in this information? But we could also just to more keywords then, sure.

Reply:
I'm out of ideas so let's maybe just stick with what you prefer.

Reply:
What about `objective_type`?

We could either let `objective` be a nice keyword that `objective_type` defaults to (that way the outer one is nicer but the inner one is not confusing) – or do `objective_type` on both. I feel that `:Riemannian` specifies a bit more the type of objective (than its domain).

Reply:
`objective_type` sounds fine :+1:

Reply:
Perfect, will change that here. Then first focus on ManifoldDiff, then Manifolds and then only return to this one here.

I think I really now understood most of that conversion stuff (just maybe not yet whether it works as well for non-isometrically embedded ones, we will see ;))